packages/datastore/src/datastore/datastore.ts

```diff
@@ -488,27 +488,26 @@ const remove: {
 		return deleted;
 	}
 };
-
 const observe: {
-	(): Observable<SubscriptionMessage<any>>;
-	<T extends PersistentModel>(obj: T): Observable<SubscriptionMessage<T>>;
-	<T extends PersistentModel>(
-		modelConstructor: PersistentModelConstructor<T>,
-		id: string
-	): Observable<SubscriptionMessage<T>>;
-	<T extends PersistentModel>(
-		modelConstructor: PersistentModelConstructor<T>
-	): Observable<SubscriptionMessage<T>>;
+	(): Observable<SubscriptionMessage<PersistentModel>>;
+
+	<T extends PersistentModel>(model: T): Observable<SubscriptionMessage<T>>;
+
 	<T extends PersistentModel>(
 		modelConstructor: PersistentModelConstructor<T>,
-		criteria: ProducerModelPredicate<T>
+		criteria?: string | ProducerModelPredicate<T>
 	): Observable<SubscriptionMessage<T>>;
-} = <T extends PersistentModel>(
-	modelConstructor?: PersistentModelConstructor<T>,
+} = <T extends PersistentModel = PersistentModel>(
+	modelOrConstructor?: PersistentModelConstructor<T>,
```

Comment:
Is missing `PersistentModel` type here?
```suggestion
	modelOrConstructor?: T | PersistentModelConstructor<T>,
```